### DISCUSSION TITLE
Select python version in windows-2019 runner agent?

### DISCUSSION THREAD

**Discussion by mpconte** [actions/virtual-environments#5260] [General questions]
I noticed that in the windows-2019 agent there are a list of python versions with 3.7.9 set to default. Is it possible to assign a different default python version without the need to use the setup-python action?

The issue I'm running into is that I'm trying to run a CMake build for boost python modules in my action but it keeps trying to use the default installed python version instead of the one I want to use that I tried to setup using the setup-python action (3.8)

**Comment by mikhailkoliada** [comment]
@mpconte Hi!

3.7.9 is in the PATH variable by default, if you call the `$env:PATH` command in your workflow you are going to see the `C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts;` amongst many others, therefore you can change the PATH the way you wish and be fine without the `setup-python` task as `setup-python` in fact modifies `$env:PATH` too.

**Comment by mpconte** [comment]
@mikhailkoliada Thanks for your response.

From what I gather, there's no real means of selecting the version of python you want without mangling the PATH variable?

**Reply by mikhailkoliada** [reply]
@mpconte yes, you are right
